packages/grid/x-grid/src/tests/events.XGrid.test.tsx

```diff
@@ -0,0 +1,267 @@
+import * as React from 'react';
+import {
+  createClientRenderStrictMode,
+  // @ts-ignore
+  fireEvent,
+} from 'test/utils';
+import { expect } from 'chai';
+import {
+  XGrid,
+  GridColParams,
+  useGridApiRef,
+  XGridProps,
+  GridRowParams,
+  GridCellParams,
+  GridRowsProp,
+  GridColumns,
+} from '@material-ui/x-grid';
+import { getCell, getColumnHeaderCell, getRow } from 'test/utils/helperFn';
+
+describe('<XGrid /> - Events Params ', () => {
+  // TODO v5: replace with createClientRender
+  const render = createClientRenderStrictMode();
+
+  const baselineProps: { rows: GridRowsProp; columns: GridColumns } = {
+    rows: [
+      {
+        id: 1,
+        first: 'Mike',
+        age: 11,
+      },
+      {
+        id: 2,
+        first: 'Jack',
+        age: 11,
+      },
+      {
+        id: 3,
+        first: 'Mike',
+        age: 20,
+      },
+    ],
+    columns: [
+      { field: 'id' },
+      { field: 'first', editable: true },
+      { field: 'age' },
+      {
+        field: 'firstAge',
+        valueGetter: (params) => `${params.row.first}_${params.row.age}`,
+        valueFormatter: (params) => `${params.value} yrs`,
+      },
+    ],
+  };
+
+  before(function beforeHook() {
+    if (/jsdom/.test(window.navigator.userAgent)) {
+      // Need layouting
+      this.skip();
+    }
+  });
+
+  let apiRef;
+  const TestEvents = (props: Partial<XGridProps>) => {
+    apiRef = useGridApiRef();
+    return (
+      <div style={{ width: 300, height: 300 }}>
+        <XGrid apiRef={apiRef} {...baselineProps} {...props} />
+      </div>
+    );
+  };
+
+  describe('columnHeaderParams', () => {
+    let eventArgs: { params: GridColParams; event: React.MouseEvent } | null = null;
```

Comment:
a spy might have been simpler. We usually use them to test callbacks

Reply:
Not a big fan of spies, I think it brings a bit more complexity with its api...
The test will lead to the same result.

Reply:
Ok, at least we use an array push now, which seems better (no forced casting to string).